compiler/src/formatting/fmt.re

```diff
@@ -0,0 +1,3252 @@
+    | PExpId({txt: Identifier.IdentName({txt: op})}) =>
+      if (String.length(op) > 1) {
+        switch (String.sub(op, 0, 2)) {
+        | "<<"
+        | ">>"
+        | ">>>"
+        | "++" => true
+        | _ => false
+        };
+      } else {
+        false;
+      }
+    | _ => false
+    };
+
+  let is_logic_op = expr =>
+    switch (expr.pexp_desc) {
+    | PExpId({txt: Identifier.IdentName({txt: op})}) =>
+      if (String.length(op) > 1) {
+        switch (String.sub(op, 0, 2)) {
+        | "<="
+        | ">="
+        | "=="
+        | "!="
+        | "is"
+        | "isnt"
+        | "&&"
+        | "||" => true
+        | _ => false
+        };
+      } else {
+        false;
+      }
+    | _ => false
+    };
+
+  let is_math_op = expr =>
+    if (is_logic_op(expr) || is_shift_or_concat_op(expr)) {
+      false;
+    } else {
+      switch (expr.pexp_desc) {
+      | PExpId({txt: Identifier.IdentName({txt: op})}) =>
+        if (String.length(op) > 0) {
+          switch (op.[0]) {
+          | '*'
+          | '/'
+          | '%'
+          | '+'
+          | '-'
+          | '<'
+          | '>'
+          | '&'
+          | '^'
+          | '|' => true
+          | _ => false
+          };
+        } else {
+          false;
+        }
+      | _ => false
+      };
+    };
+
+  let op_precedence = startsWith =>
+    switch (startsWith) {
+    | '*'
+    | '/'
+    | '%' => 120
+    | '+'
+    | '-' => 110
+    | '<'
+    | '>' => 90
+    | '&' => 70
+    | '^' => 60
+    | '|' => 50
+    | '_' => 10
+    | _ => 9999
+    };
+
+  let precedence = expr => {
+    switch (expr.pexp_desc) {
+    | PExpId({txt: Identifier.IdentName({txt: op})}) =>
+      if (String.length(op) > 1) {
+        switch (String.sub(op, 0, 2)) {
+        | "++" => 110
+        | "<<"
+        | ">>" => 100
+        | "=="
+        | "!="
+        | "is" => 80
+        | "&&" => 40
+        | "||"
+        | "??" => 30
+        | _ => op_precedence(op.[0])
+        };
+      } else if (String.length(op) > 0) {
+        op_precedence(op.[0]);
+      } else {
+        9999;
+      }
+    | _ => 9999
+    };
+  };
+
+  let infixop = op => {
+    switch (op.[0]) {
+    | '+'
+    | '-'
+    | '*'
+    | '/'
+    | '%'
+    | '='
+    | '^'
+    | '<'
+    | '>'
+    | '&'
+    | '|'
+    | '?' => true
+    | _ when op == "is" => true
+    | _ when op == "isnt" => true
+    | _ when String.starts_with(~prefix="!=", op) => true
+    | _
+    | exception _ => false
+    };
+  };
+
+  let is_infix_op = expr => {
+    switch (expr.pexp_desc) {
+    | PExpId({txt: Identifier.IdentName({txt: op})}) => infixop(op)
+    | _ => false
+    };
+  };
+
+  let prefixop = op =>
+    switch (op.[0]) {
+    | '!' => true
+    | _
+    | exception _ => false
+    };
+
+  let is_prefix_op = expr => {
+    switch (expr.pexp_desc) {
+    | PExpId({txt: Identifier.IdentName({txt: op})}) => prefixop(op)
+    | _ => false
+    };
+  };
+
+  let is_keyword = expr => {
```

Comment:
This function name seems a little misleading; to me this implies it would catch any keyword in the language. Maybe `is_keywordlike_function`?